internal/assets/manifests/istio-meshgateway/templates/deployment.yaml

```diff
@@ -48,6 +48,12 @@ spec:
             - containerPort: {{ $val.targetPort | default $val.port }}
               protocol: {{ $val.protocol | default "TCP" }}
             {{- end }}
+            - containerPort: 15020
+              protocol: TCP
+              name: http-merged-prom
+            - containerPort: 15021
+              protocol: TCP
+              name: http-health-check
```

Comment:
If the user would like to expose port 15021 publicly, so on the `service` resource as well, then they should add this port to their `IstioMeshGateway` resource spec. If they do that, the 15021 will be appended to the `containerPorts` as well, as you can see a few lines above (around 48) in this file. But if we add port 15021 explicitly as you propose in this PR, then it would be duplicated. That's why I suggest **not to add 15021 here**, only 15020 as it is most likely not intended to be exposed publicly (on the `service`).

Reply:
Sorry, I might be wrong here, but exposing it on the service would IMHO need an explicit
```
  ports:
  [...]
    - protocol: TCP
      port: 15021
      targetPort: 15021
```
to work, wouldn't it?

Reply:
Yes,  this should be added to the `IstioMeshGateway` spec if you'd like to expose this port on the service.

Reply:
I did not mean to expose the port to the outside of the namespace network. But since this is a health check port, it might be really handy to expose the container port.